### PR TITLE
User/xianz/depend shared lib

### PR DESCRIPTION
**Pull request by zhangxiang1993**
This PR is to add dependencies of all samples on sharedLib, so the sharelib will be built before each sample build.

### CODE REVIEW

File: Samples/CustomOperator/desktop/cpp/operators/GeneratedShaders/relu_shader.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ dcl_thread_group 64, 1, 1
 // Initial variable locations:
 //   vThreadID.x <- dtid.x; vThreadID.y <- dtid.y; vThreadID.z <- dtid.z
 //
-#line 13 "C:\samples\Windows-Machine-Learning\Samples\CustomOperator\desktop\cpp\operators\relu_shader.hlsl"
+#line 13 "E:\xianz\Windows-Machine-Learning\Samples\CustomOperator\desktop\cpp\operators\relu_shader.hlsl"
```
Comment (martinb35) [inline Samples/CustomOperator/desktop/cpp/operators/GeneratedShaders/relu_shader.h:64]: this change looks unintentional - do you want it?

File: Samples/CustomOperator/desktop/cpp/operators/GeneratedShaders/relu_shader.h [hunk 2]
```diff
@@ -1043,93 +1043,93 @@ const BYTE g_Relu[] =
...
+     44,  32,  49,  44,  32,  49, 
+     41,  93,  13,  10, 118, 111, 
+    105, 100,  32,  82, 101, 108, 
+    117,  40, 117, 105, 110, 116, 
+     51,  32, 100, 116, 105, 100, 
+     32,  58,  32,  83,  86,  95, 
+     68, 105, 115, 112,  97, 116, 
+     99, 104,  84, 104, 114, 101, 
+     97, 100,  73, 100,  41,  13, 
      10, 123,  13,  10,  32,  32,
```
Comment (martinb35) [inline Samples/CustomOperator/desktop/cpp/operators/GeneratedShaders/relu_shader.h:1120]: why did the shader change?

File: Samples/CustomOperator/desktop/cpp/operators/GeneratedShaders/relu_shader.h (hunk 1 above)
Comment (tiagoshibata) [inline Samples/CustomOperator/desktop/cpp/operators/GeneratedShaders/relu_shader.h:64]: Nit: machine specific path

Why did the shader change?

Comment (tiagoshibata) [reply Samples/CustomOperator/desktop/cpp/operators/GeneratedShaders/relu_shader.h:64]: Brian beat me by a minute 😄

**Pull request merged**